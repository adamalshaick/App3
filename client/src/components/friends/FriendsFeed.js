import React, { Component } from "react";
import { Link } from "react-router-dom";
import ProfileItem from "../profiles/ProfileItem";

class FriendsFeed extends Component {
  render() {
    const { profile, profiles } = this.props;
    let friendItems;

    if (profile.friends && profile.friends.length) {
      const friends = profiles.filter(friend =>
        profile.friends.includes(friend.user._id)
      );
      friendItems = friends.map(friend => (
        <ProfileItem key={friend._id} profile={friend} />
      ));
    } else {
      friendItems = (
        <div className="text-center">
          <p>
            <i>You didn't add any friend yet</i>
          </p>
          <Link to="/feed" className="btn btn-outline-dark">
            Browse Profiles
          </Link>
        </div>
      );
    }

    return <div className="row">{friendItems}</div>;
  }
}

export default FriendsFeed;
