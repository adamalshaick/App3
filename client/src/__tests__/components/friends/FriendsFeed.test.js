import React from "react";
import FriendsFeed from "../../../components/friends/FriendsFeed";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

const Props = {
  profile: {
    friendRequests: [{ _id: "id" }, { _id: "id" }]
  },

  profiles: {}
};

wrapper = shallow(<FriendsFeed {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
