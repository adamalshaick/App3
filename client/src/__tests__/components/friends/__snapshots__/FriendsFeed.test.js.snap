// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="row"
>
  <div
    className="text-center"
  >
    <p>
      <i>
        You didn't add any friend yet
      </i>
    </p>
    <Link
      className="btn btn-outline-dark"
      replace={false}
      to="/feed"
    >
      Browse Profiles
    </Link>
  </div>
</div>
`;
